kedro/pipeline/node.py

```diff
@@ -281,7 +281,7 @@ def name(self) -> str:
         Returns:
             Node's name if provided or the name of its function.
         """
-        node_name = self._name or str(self)
+        node_name = self._name or self._func_name
```

Comment:
IIUC this is one of the key bits, right? `name` is a calculated property, and instead of using `__str__` below, it's using `._func_name` in the absence of a given `._name`

Reply:
This is the key bit for the second proposal in this PR as described in the PR description. This part makes it possible to use the function name to filter a run by a specific node.